File: leetcode/lc-us/go/1041.go
package main

func isRobotBounded(instructions string) bool {
	dx := []int{1, 0, -1, 0}
	dy := []int{0, 1, 0, -1}
	x, y, d := 0, 0, 0
	for i := 0; i < len(instructions); i++ {
		switch instructions[i] {
		case 'L':
			d = (d + 1) % 4
			break
		case 'R':
			d = (d + 3) % 4
			break
		default:
			x += dx[d]
			y += dy[d]
		}
	}
	return x == 0 && y == 0 || d != 0
}

File: leetcode/lc-us/rust/1041.rs
impl Solution {
    pub fn is_robot_bounded(instructions: String) -> bool {
        let dx = [1, 0, -1, 0];
        let dy = [0, 1, 0, -1];
        let mut x = 0;
        let mut y = 0;
        let mut d = 0;
        for c in instructions.chars() {
            match c {
                'L' => { d = (d + 1) % 4 }
                'R' => { d = (d + 3) % 4 }
                'G' => {
                    x += dx[d];
                    y += dy[d];
                }
                _ => {}
            }
        }
        x == 0 && y == 0 || d != 0
    }
}